### PR TITLE
[Driver] SR-3175: Include the terminating signal number in the driver output

### PR DESCRIPTION
**Pull request by matthewcarroll**
* Add the signal number of the terminated task to the output of the driver on platforms for which the signal number is available. The new key in the parseable driver output is "signal".
* Add a test to verify that the signal number is emitted.
* Add documentation for the new "signal" key emitted in the parseable driver output.

https://bugs.swift.org/browse/SR-3175

In this commit, example output of swiftc -Xfrontend -debug-crash-immediately is:
```
{
"kind": "signalled",
"name": "compile",
"pid": 32822,
"output": "0 swift ...

"error-message": "Illegal instruction: 4",
"signal": 4
<unknown>:0: error: unable to execute command: Illegal instruction: 4
<unknown>:0: error: compile command failed due to signal 4 (use -v to see invocation)
}
```

### CODE REVIEW

File: lib/Driver/Compilation.cpp [hunk 1]
```diff
@@ -636,9 +636,15 @@ int Compilation::performJobsImpl() {
...
                      ErrorMsg);
-
-    Diags.diagnose(SourceLoc(), diag::error_command_signalled,
-                   SignalledCmd->getSource().getClassName());
+    
+    if (Signal.hasValue()) {
+      Diags.diagnose(SourceLoc(), diag::error_command_signalled,
+                     SignalledCmd->getSource().getClassName(), Signal.getValue());
+    }
+    else {
```
Comment (CodaFi) [inline lib/Driver/Compilation.cpp]: Nit: `else` in line with `}` please

File: lib/Driver/ParseableOutput.cpp [hunk 2]
```diff
@@ -201,15 +201,17 @@ class FinishedMessage : public TaskOutputMessage {
 
 class SignalledMessage : public TaskOutputMessage {
   std::string ErrorMsg;
+  Optional<int> Signal;
 public:
   SignalledMessage(const Job &Cmd, ProcessId Pid, StringRef Output,
-                   StringRef ErrorMsg) : TaskOutputMessage("signalled", Cmd,
+                   StringRef ErrorMsg, Optional<int> Signal) : TaskOutputMessage("signalled", Cmd,
```
Comment (xwu) [inline lib/Driver/ParseableOutput.cpp]: Nit: with the additional parameter, the alignment of the two lines that follow are totally out of whack.

**Comment by CodaFi** [comment]
@swift-ci please smoke test.

**Comment by CodaFi** [comment]
Whoa, an NPE took down the build!   // @shahmishal 

@swift please smoke test OS X platform.

**Comment by CodaFi** [comment]
@swift-ci please smoke test OS X platform.

**Comment by matthewcarroll** [comment]
@CodaFi,

I merged the branch `hughbe:taskqueue-fixes` from #6812

**Comment by CodaFi** [comment]
Can you knock off the merge commit either by folding it into the previous one or by cherry-picking @hughbe's branch please?

**Comment by shahmishal** [comment]
@swift-ci please smoke test.

**Comment by hughbe** [comment]
@swift-ci please smoke test.

**Comment by hughbe** [comment]
Is this PR ready to go? Merging will unblock the Windows (clang and MSVC) build

**Comment by matthewcarroll** [comment]
@hughbe Yes it's ready.

@CodaFi What do you think?

**Comment by CodaFi** [comment]
⛵️

**Pull request merged**